adapter/postgres/postgres.go

```diff
@@ -345,19 +352,30 @@ func (d *PostgresDatabase) getColumns(table string) ([]column, error) {
 }
 
 func (d *PostgresDatabase) getIndexDefs(table string) ([]string, error) {
-	const query = "SELECT indexName, indexdef FROM pg_indexes WHERE schemaname=$1 AND tablename=$2"
+	// Exclude indexes that are implicitly created for primary keys or unique constraints.
+	const query = `WITH
+	  unique_and_pk_constraints AS (
+	    SELECT con.conname AS name
+	    FROM   pg_constraint con
+	    JOIN   pg_namespace nsp ON nsp.oid = con.connamespace
+	    JOIN   pg_class cls ON cls.oid = con.conrelid
+	    WHERE  con.contype IN ('p', 'u')
+	    AND    nsp.nspname = $1
+	    AND    cls.relname = $2
+	  )
+	SELECT indexName, indexdef
+	FROM   pg_indexes
+	WHERE  schemaname = $1
+	AND    tablename = $2
+	AND    indexName NOT IN (SELECT name FROM unique_and_pk_constraints)
+	`
 	schema, table := SplitTableName(table)
 	rows, err := d.db.Query(query, schema, table)
 	if err != nil {
 		return nil, err
 	}
 	defer rows.Close()
 
-	uniqueConstraints, err := d.getUniqueConstraints(table)
```

Comment:
Since `rows.Close()` is deferred and has not been executed yet at this line, `d.getUniqueConstraints(table)` always returns an empty slice unexpectedly.